src/Equinox.MessageDb/MessageDbClient.fs

```diff
@@ -98,23 +115,17 @@ type MessageDbClient(source: CancellationToken -> Task<NpgsqlConnection>) =
 
         cmd
 
-    member __.WriteMessages(streamName, events : _ array, version : int64, ct) =
-        task {
-            try
-                use! conn = source ct
-                use transaction = conn.BeginTransaction()
-                use batch = new NpgsqlBatch(conn, transaction)
-
-                let mutable expectedVersion = version
-
-                for event in events do
-                    batch.BatchCommands.Add(__.PrepareWriteCommand(streamName, expectedVersion, event))
+    member __.WriteMessages(streamName, events : _ array, version : int64, ct) = task {
+        try
+            use! conn = source ct
+            use transaction = conn.BeginTransaction()
+            use batch = new NpgsqlBatch(conn, transaction)
 
-                    expectedVersion <- expectedVersion + 1L
+            let prep i event = batch.BatchCommands.Add(__.PrepareWriteCommand(streamName, version + int64 i, event))
```

Comment:
```suggestion
            let prep i event = x.PrepareWriteCommand(streamName, version + int64 i, event)
            events |> Array.iteri (prep >> batch.BatchCommands.Add)
```

Reply:
I wish... `Add` has an overload so the compiler chokes out 😞